packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,527 @@
+        :min-date="minDate"
+        :mode="mode"
+        :model-config="modelConfig"
+        :select-attribute="calendarSelectAttributes"
+        is-expanded
+      />
+      <div
+        v-else-if="hasRelativeTimeframes && hasTimePeriods"
+        class="d-flex flex-column"
+      >
+        <div
+          v-for="(item, index) in timePeriods"
+          :key="`section-${index}`"
+          class="timeframe-section d-flex flex-column"
+        >
+          <h4>{{ item.section }}</h4>
+          <div class="timeframe-buttons d-flex flex-row justify-content-start">
+            <KButton
+              v-for="(timeFrame, index) in item.values"
+              :key="`time-${index}`"
+              :is-rounded="false"
+              :class="{'selected-option': timeFrame.key === selectedTimeframe.key}"
+              :data-testid="'select-timeframe-' + timeFrame.timeframeLength()"
+              appearance="outline"
+              size="medium"
+              @click="changeRelativeTimeframe(timeFrame)"
+            >
+              {{ ucWord(timeFrame.timeframeText) }}
+            </KButton>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template
+      v-if="!hidePopover"
+      #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :disabled="submitDisabled"
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { format } from 'date-fns'
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Select a time range'
+    },
+    mode: {
+      type: String,
+      required: false,
+      default: '',
+      validator: (value) => {
+        return [ '', 'date', 'time', 'dateTime' ].indexOf(value) !== -1
+      }
+    },
+    defaultRelative: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
+    defaultCustom: {
+      type: Object | String,
+      required: false,
+      default: ''
+    },
+    range: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    maxDate: {
+      type: Number,
+      required: false,
+      default: null
+    },
+    minDate: {
+      type: Number,
+      required: false,
+      default: null
+    },
```

Comment:
Is `Number` the right prop type here for minDate and maxDate?

Reply:
I can't tell if this is a trick question :)

`new Date().getTime()` seems to return a number, so...